### PR TITLE
Fix text rendering where hidpi != 1

### PR DESCRIPTION
**Pull request by Xaeroxe**
What it says on the tin.

Enjoy @azriel91 !

### CODE REVIEW

**Comment by azriel91** [comment]
Yay! Thank you :bowing_man:!

File: amethyst_renderer/src/resources.rs [hunk 1]
```diff
@@ -126,4 +126,9 @@ impl ScreenDimensions {
         self.aspect_ratio = w as f32 / h as f32;
         self.dirty = true;
     }
+
+    /// Updates the hidpi factor stored in this structure.
+    pub(crate) fn update_hidpi_factor(&mut self, factor: f64) {
```
[the remark this answers is not in the document]
Comment (jojolepro) [reply amethyst_renderer/src/resources.rs]: thanks

**Comment by azriel91** [comment]
Woosh, the rendering is all fixed :tada:. Tried out the UI example, there's still a conversion that hasn't happened between the mouse position and text edit glyph position. Y position seems to be correct, but when you click (and drag), it's a bit too far left for a UiText on the left side of the screen.

**Comment by Xaeroxe** [comment]
Oh, good catch thanks!  I got so focused on keyboard stuff I totally forgot about the mouse selection.

**Comment by jojolepro** [comment]
ready for merge?

**Comment by Xaeroxe** [comment]
Yup

bors r+

**Pull request merged**